File: src/sorting/insertion_sort/insertion_sort_test.py
from insertion_sort import sort
import unittest


class InsertionSortTest(unittest.TestCase):
    def test(self):
        arr = [5, 4, 3, 2, 1]
        arr2 = [5, 2, 4, 6, 1, 3]
        sort(arr)
        sort(arr2)
        self.assertEqual(arr, [1, 2, 3, 4, 5])
        self.assertEqual(arr2, [1, 2, 3, 4, 5, 6])


if __name__ == "__main__":
    unittest.main()

File: src/sorting/insertion_sort/insertion_sort.test.ts
import { sort } from "./insertion_sort";

describe("Insertion Sort", () => {
  test("Should sort the list of numbers in ascending order", () => {
    const arr = [5, 4, 3, 2, 1];
    sort(arr);
    expect(arr).toEqual([1, 2, 3, 4, 5]);

    const arr2 = [5, 2, 4, 6, 1, 3];
    sort(arr2);
    expect(arr2).toEqual([1, 2, 3, 4, 5, 6]);
  });
});
